### ISSUE TITLE
Input width and height must be equal

### ISSUE THREAD

**Issue by liuwei16** [felixlaumon/deform-conv#10]
If the width and height have different size, there will be incompatiable error,  this should be improved  in your code for better adapation.

**Comment by 7LFB** [comment]
I encountered  the same error when I want to implement this deform module with different width and height,  do you have any idea to do with it? @liuwei16 @felixlaumon

**Comment by 7LFB** [comment]
thanks for your warning!  I modify the grid dimension setting in  'deform_conv.py'.

**Comment by mikigom** [comment]
@7LFB Could you tell me exactly how it was modified?

**Comment by 7LFB** [comment]
In the definition of function "tf_batch_map_offsets(input, offsets, order=1)":

grid = tf.meshgrid(
         tf.range(input_size), tf.range(input_size), indexing='ij'
     )

is modified into

grid = tf.meshgrid(
        tf.range(input_size_h), tf.range(input_size_w), indexing='ij'
     )

hope this helps. @mikigom

**Comment by mikigom** [comment]
@7LFB Thank you very much! It would help me a lot.

**Comment by Manyunze** [comment]
I'm afraid simply modifying grid is not enough.

coords = tf.clip_by_value(coords, 0, tf.cast(input_size, 'float32') - 1)
in tf_batch_map_coordinates clip both direction with the input_size, which is input_shape[1].

Maybe this should also be modified?

**Comment by liuwei16** [comment]
@Manyunze @7LFB Yes, in tf_batch_map_coordinates, input size should also be modified accordingly:
coords_h = tf.clip_by_value(coords_h, 0, tf.cast(input_size_h, 'float32') - 1)
coords_w = tf.clip_by_value(coords_w, 0, tf.cast(input_size_w, 'float32') - 1)
coords = tf.stack([coords_h, coords_w], axis=-1)

**Comment by liuwei16** [comment]
@7LFB  The filter size in ConvOffset2D is set as (3,3), which exhausts much computational burden when the channels of input feature maps is large (e.g. 1024, 2048).
How about changing it into (1,1)?
What is the implementation in the original caffe version?
Thank you very much.

**Comment by Manyunze** [comment]
@liuwei16 Strange thing is that, by using this kind of implementation of deformable conv, my memory usage keeps increasing in a strangely high speed, after 2 to 3 epochs, the process will be killed by OS because of OOM.
